Source/Core/Core/DSP/DSPAssembler.cpp

```diff
@@ -425,33 +425,38 @@ u32 DSPAssembler::GetParams(char* parstr, param_t* par)
   return count;
 }
 
-const opc_t* DSPAssembler::FindOpcode(const char* name, u32 par_count, const opc_t* const opcodes,
-                                      size_t opcodes_size)
+const opc_t* DSPAssembler::FindOpcode(std::string name, size_t par_count, OpcodeType type)
 {
   if (name[0] == 'C' && name[1] == 'W')
     return &cw;
 
   const auto alias_iter = aliases.find(name);
   if (alias_iter != aliases.end())
-    name = alias_iter->second.c_str();
-  for (size_t i = 0; i < opcodes_size; i++)
+    name = alias_iter->second;
+
+  const DSPOPCTemplate* const info = [type, &name] {
+    if (type == OpcodeType::Primary)
+      return FindOpInfoByName(name);
+
+    return FindExtOpInfoByName(name);
+  }();
```

Comment:
oh, please no. Is the const as important here? If so:
const DSPOPCTemplate* const info = type == OpcodeType::Primary ? FindOpInfoByName(name) : FindExtOpInfoByName(name);